### ISSUE TITLE
Linux client stuck in memory

### ISSUE THREAD

**Issue by alexander-potemkin** [mattermost/desktop#1844]
I confirm (by marking "x" in the [ ] below: [x]):

- [ x ] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [ x ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [ x ] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
Mattermost 5.0.1 stuck in memory.

**Environment**
- Operating System: Linux Mint 20.2
- Mattermost Desktop App version: 5.0.1
- Mattermost Server version: 6.0.2

**Steps to reproduce**

Close Mattermost
Open it again

**Expected behavior**

Mattermost to close and then to open.

**Observed behavior**

It doesn't start.

**Possible fixes**

Killing all stuck process.

![mattermost-dekstop](https://user-images.githubusercontent.com/2323840/139445263-801b50c5-75be-4ca5-9cf0-15b58aa5068f.png)

**Comment by alexander-potemkin** [comment]
Thank you. I had to add tray icon as a work-around, indeed.

I’m not sure I’ve got about Ubuntu specific app closing.

btw, Ubuntu is quite popular foundation for many distros, so I would say it’s quite a thing to address, unless I’m missing something?

**Comment by devinbinnie** [comment]
@alexander-potemkin On Ubuntu's default configuration for GNOME, there's a system tray in the top bar. I know other distros like Fedora don't use this, as well as Ubuntu-based distros that don't use GNOME like Linux Mint. The app is built to use that tray just like Windows and Mac do, but we're aware that a lot of other distros don't use that.

tl;dr Our Linux closing behaviour works for a lot of Ubuntu users, but not all of them and it doesn't work for a lot of other distros so we need to address that. There's a JIRA ticket floating around for that somewhere.

**Comment by alexander-potemkin** [comment]
Hm, got it. So it's not that much about Linux distribution, as it's about tray icon behaviour, got it, thanks!
Yes, fixing this would be very nice. It's also wasn't the problem before 5.x client.

**Comment by alexander-potemkin** [comment]
@devinbinnie there is one major difference in how Mattermost restores from the tray vs Slack or Telegram: when I click on tray icon on Mattermost, I have to choose which workspace I would like to have opened; while on other messengers (and apps) - I get app window immediately, which is muuuuch more convenient.

**Comment by alexander-potemkin** [comment]
on top of that, on Slack I can click and restore window via normal icon and via tray icon, so tray icon is a way to stay at background.

**Comment by devinbinnie** [comment]
> @devinbinnie there is one major difference in how Mattermost restores from the tray vs Slack or Telegram: when I click on tray icon on Mattermost, I have to choose which workspace I would like to have opened; while on other messengers (and apps) - I get app window immediately, which is muuuuch more convenient.

To clarify, do you mean which desktop workspace this opens on?

> on top of that, on Slack I can click and restore window via normal icon and via tray icon, so tray icon is a way to stay at background.

This is being fixed here: https://github.com/mattermost/desktop/pull/1848

**Comment by alexander-potemkin** [comment]
> To clarify, do you mean which desktop workspace this opens on?

By workspace I mean, not a desktop workspace - virtual desktops, I meant Mattermost workspace / chats I'm connected to (I have different servers I'm connected to).

**Comment by devinbinnie** [comment]
> > To clarify, do you mean which desktop workspace this opens on?
> 
> By workspace I mean, not a desktop workspace - virtual desktops, I meant Mattermost workspace / chats I'm connected to (I have different servers I'm connected to).

I see what you mean. On other OSes we're able to double-click on the tray icon to just open the window. Apparently on Linux (or Ubuntu at least) you can't. I think the same workflow should be possible using the task bar once the above fix mentioned is merged.

**Comment by alexander-potemkin** [comment]
Yes, please

**Comment by devinbinnie** [comment]
JIRA ticket here: https://mattermost.atlassian.net/browse/MM-30713

**Comment by devinbinnie** [comment]
Fixed in this PR: https://github.com/mattermost/desktop/pull/1988
